Retire standalone C# unions article; document where unions aren't supported (#37276)
dotnet/AspNetCore.Docs: aspnetcore/blazor/components/index.md

If the component parameter is of type string, then the attribute value is instead treated as a C# string literal. If you want to specify a C# expression instead, then use the `@` prefix.

:::moniker range=">= aspnetcore-11.0"

> The string-literal shortcut applies only to parameters declared as `string`. A parameter declared as a [C# union type](/dotnet/csharp/whats-new/csharp-14#union-types) — even one whose cases include `string` — isn't a `string`-typed parameter, so the attribute value must be a C# expression. Use the `@` prefix, for example `Message="@("Saved.")"`, tracked at [dotnet/razor#13188](https://github.com/dotnet/razor/issues/13188).

Reviewer: ```suggestion
The string-literal shortcut applies only to parameters declared as `string`. A parameter declared as a [C# union type](/dotnet/csharp/whats-new/csharp-14#union-types), even one whose cases include `string`, isn't a `string`-typed parameter, so the attribute value must be a C# expression. Use the `@` prefix, for example `Message="@("Saved.")"`. This requirement is tracked at [dotnet/razor#13188](https://github.com/dotnet/razor/issues/13188).
```
Very minor: Split to two sentances to make this a little more readable.

Reviewer 2: If for updates at .NET 11, lead the paragraph with this ...

```html
<!-- UPDATE 11.0 - Remove the following paragraph per resolution of the PU issue -->
```

... and for the link convention, use this format ...

```diff
- [dotnet/razor#13188](https://github.com/dotnet/razor/issues/13188)
+ [Razor: extend literal-attribute shortcut to component parameters typed as a C# union with a string case (`dotnet/razor` #13188)](https://github.com/dotnet/razor/issues/13188)
```

Here's a full suggestion for these, including Wade's suggestions. The issue is about making a change to the framework to address this, so I changed "requirement" to our 'for more info' lead-in ...

```suggestion
<!-- UPDATE 11.0 - Remove the following paragraph per resolution of the PU issue -->

The string-literal shortcut applies only to parameters declared as `string`. A parameter declared as a [C# union type](/dotnet/csharp/whats-new/csharp-14#union-types), even one whose cases include `string`, isn't a `string`-typed parameter, so the attribute value must be a C# expression. Use the `@` prefix, for example `Message="@("Saved.")"`. For more information, see [Razor: extend literal-attribute shortcut to component parameters typed as a C# union with a string case (`dotnet/razor` #13188)](https://github.com/dotnet/razor/issues/13188).
```

If the component parameter is of type string, then the attribute value is instead treated as a C# string literal. If you want to specify a C# expression instead, then use the `@` prefix.

:::moniker range=">= aspnetcore-11.0"

<!-- UPDATE 11.0 - Remove the following paragraph per resolution of the PU issue -->

> The string-literal shortcut applies only to parameters declared as `string`. A parameter declared as a [C# union type](/dotnet/csharp/whats-new/csharp-14#union-types), even one whose cases include `string`, isn't a `string`-typed parameter, so the attribute value must be a C# expression. Use the `@` prefix, for example `Message="@("Saved.")"`. For more information, see [Razor: extend literal-attribute shortcut to component parameters typed as a C# union with a string case (`dotnet/razor` #13188)](https://github.com/dotnet/razor/issues/13188).

Author: ```suggestion
The string-literal shortcut applies only to parameters declared as `string`. A parameter declared as a [C# union type](/dotnet/csharp/whats-new/csharp-14#union-types), even one whose cases include `string`, isn't a `string`-typed parameter, so the attribute value must be a C# expression. Use the `@` prefix, for example `Message="@("Saved.")"`.
```
